### PR TITLE
Improve parsing recovery with 'file' keyword

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/75992

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler @RikkiGibson ptal.

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs [hunk 1]
```diff
@@ -2054,39 +2054,57 @@ private bool IsPossibleAggregateClauseStartOrStop()
...
+            list.Add(this.CurrentToken.Kind == SyntaxKind.OpenParenToken
+                ? _syntaxFactory.PrimaryConstructorBaseType(firstType, this.ParseParenthesizedArgumentList())
                 : _syntaxFactory.SimpleBaseType(firstType));
 
-            // any additional types
+            // Parse any optional base types that follow.
             while (true)
             {
-                if (this.CurrentToken.Kind == SyntaxKind.OpenBraceToken ||
-                    ((_termState & TerminatorState.IsEndOfRecordOrClassOrStructOrInterfaceSignature) != 0 && this.CurrentToken.Kind == SyntaxKind.SemicolonToken) ||
```
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs:2078]: _termState check was unnecessary.  Assert and comment added at start of method to explain why.

File: src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs [hunk 2]
```diff
@@ -3598,4 +3599,106 @@ void M(file x) { }
         }
         EOF();
     }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/75992")]
+    public void TestFileModifierAfterIncompleteBaseList1()
+    {
+        UsingTree("""
+            class C : B
+            file class D
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs:3608]: I imagine a similar, more contrived case may also be positively impacted

```cs
class Outer
{
    class C : B

    required int Prop { get; set; }
}
```

but, don't find it interesting enough to ask you to come back and add the test.

**Review by RikkiGibson — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler for a second pair of eyes on this simple pr :)  Thanks!

**Review by 333fred — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**